demo/external-player.html

```diff
@@ -0,0 +1,256 @@
+        to allow audio editors to make accurate clippings of audio content.
+        You can read more about the project
+        <a href="https://waveform.prototyping.bbc.co.uk/">here</a>.
+      </p>
+
+      <h2>Demo pages</h2>
+      <p>
+        The following pages demonstrate various configuration options:
+      </p>
+      <p>
+        <a href="index.html">Precomputed Waveform Data</a> |
+        <a href="webaudio.html">Web Audio API</a> |
+        <a href="zoomable-waveform.html">Single Zoomable Waveform</a> |
+        <a href="overview-waveform.html">Single Fixed Waveform</a> |
+        <a href="cue-events.html">Cue Events</a> |
+        <a href="set-source.html">Changing the Media URL</a> |
+        <a href="multi-channel.html">Multi-Channel Waveform</a> |
+        <a href="custom-markers.html">Custom Point and Segment Markers</a> |
+        External Audio Player
+      </p>
+      <h2>Demo: External Player</h2>
+      <p>
+        This demo shows how to use an external player for audio playback with Peaks.js. A Tone.js Player is setup in the example.
+      </p>
+    </div>
+
+    <div id="waveform-container">
+      <div id="zoomview-container"></div>
+      <div id="overview-container"></div>
+    </div>
+
+    <div id="demo-controls">
+      <audio id="audio" controls="controls" hidden>
+        <source src="TOL_6min_720p_download.mp3" type="audio/mpeg">
+        <source src="TOL_6min_720p_download.ogg" type="audio/ogg">
+        Your browser does not support the audio element.
+      </audio>
+
+      <div id="controls">
+        <button data-action="play">Play</button>
+        <button data-action="pause">Pause</button>
+        <input type="text" id="seek-time" value="0.0">
+        <button data-action="seek">Seek</button>
+      </div>
+    </div>
+
+    <script src="peaks.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/tone/13.8.25/Tone.js"></script>
+
+    <script>
+      (async function(Peaks) {
+        const audioContext = Tone.context;
+
+        // Load audio file into audio buffer
+        await fetch("TOL_6min_720p_download.mp3")
+          .then(function(response) {
+            return response.arrayBuffer();
+          })
+          .then(function(buffer) {
+            return audioContext.decodeAudioData(buffer);
+          })
+          .then(function(audioBuffer) {
+            const externalPlayer = new Tone.Player(audioBuffer);
+
+            const player = {
+              init: function(internalPlayer) {
+                externalPlayer.sync();
+                externalPlayer.start();
+
+                Tone.connectSeries(externalPlayer, Tone.Master);
+
+                Tone.Transport.scheduleRepeat(time => {
+                  internalPlayer._updatedTime();
+
+                  if (internalPlayer.getCurrentTime() >= internalPlayer.getDuration()) {
+                    Tone.Transport.stop();
+                    onSongComplete();
+                  }
+                }, 0.03);
```

Comment:
Does this need to be faster than the media element's default, which is about every 250 milliseconds? Depending on zoom leve, Peaks.js will switch to using `requestAnimationFrame` and polling the player's current time.

Reply:
You're right, since this `Tone.Transport.scheduleRepeat()` is only used to inform about the current time, the 250ms default is fine. A user might also add want to add synth events at different intervals, but he will choose a different `Tone.Transport.scheduleRepeat()` then. Changed the example accordingly.